lib/libvarnishapi/vsl_query.c

```diff
@@ -144,15 +144,40 @@ vslq_test_rec(const struct vex *vex, const struct VSLC_ptr *rec)
 
 	/* Field */
 	if (vex->lhs->field > 0) {
-		for (e = b, i = 0; *e && i < vex->lhs->field; i++) {
+		for (e = b, i = 0, dq = 0; *e && i < vex->lhs->field; i++) {
+			/* Skip end of previous field */
+			if (dq) {
```

Comment:
why this block...

Reply:
This block checks what's after the closing dquote of the previous field, skips the dquote `e` is pointing to and makes sure a proper "whitespace" field delimiter follows the closing dquote.